### PR TITLE
Update Branch-2.1 for Apache License 

### PR DESCRIPTION
**Pull request by merlintang**
Manually build locally. 

Do not have the test, and will update the test later.

### CODE REVIEW

**Comment by merlintang** [comment]
@yuyongyang800  please squash and merge this commit.

**Comment by merlintang** [comment]
@yuyongyang800  please review it at first, I am not sure my update is correct.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+[![Build Status](https://travis-ci.org/merlintang/SparkDistributedMatrix.svg?branch=branch-2.1)](https://travis-ci.org/merlintang/SparkDistributedMatrix)
+
+# Apache Spark&trade; Distributed Matrix Computation
+
+An library to support distributed matrix computation for machine learning and data analysis
```
Comment (yuyongyang800) [inline README.md]: A library, not An

File: src/main/scala/org/apache/spark/sql/matfast/matrix/BLAS.scala [hunk 2]
```diff
@@ -372,7 +375,7 @@ object BLAS extends Serializable with Logging {
     if (A.isTransposed) {
       var colCounterForB = 0
       if (!B.isTransposed) { // Expensive to put the check inside the loop
-        while (colCounterForB < nB) {
+        while ( colCounterForB < nB) {
```
Comment (yuyongyang800) [inline src/main/scala/org/apache/spark/sql/matfast/matrix/BLAS.scala]: Why do we add the space here?

File: src/main/scala/org/apache/spark/sql/matfast/partitioner/ColumnPartitioner.scala [hunk 3]
```diff
@@ -21,7 +21,7 @@ import org.apache.spark.{Partitioner, SparkConf}
 import org.apache.spark.rdd.{RDD, ShuffledRDD}
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.matfast.util.MatfastSerializer
-
+// scalastyle:off
```
Comment (yuyongyang800) [inline src/main/scala/org/apache/spark/sql/matfast/partitioner/ColumnPartitioner.scala:24]: If we just add // scalastyle:off without tuning it on, then all the code below will not go through the style checking, right? Why do we skip the style checking?

**Comment by yuyongyang800** [comment]
I have updated the master branch to spark-2.1. Do you mean to send the PR to master branch or spark-2.1 branch?

File: README.md (hunk 1 above)
Comment (merlintang) [reply README.md]: ok, fix it.

File: src/main/scala/org/apache/spark/sql/matfast/matrix/BLAS.scala (hunk 2 above)
Comment (merlintang) [reply src/main/scala/org/apache/spark/sql/matfast/matrix/BLAS.scala]: fixed

File: src/main/scala/org/apache/spark/sql/matfast/partitioner/ColumnPartitioner.scala (hunk 3 above)
Comment (merlintang) [reply src/main/scala/org/apache/spark/sql/matfast/partitioner/ColumnPartitioner.scala:24]: scalastyle need to include the implement the hashcode function as well.

**Comment by merlintang** [comment]
we can keep a master branch and 2.1 branch as well. for our current pr, it can go to branch 2.1, then go to master branch. as you can notice, we need to have branch 2.2 later. and keep the branch master as the same of branch-2.2

**Review by yuyongyang800 — APPROVED** [review]

**Comment by merlintang** [comment]
thanks @yuyongyang800  can you cherry-pick this commit to master branch as well? git cherry-pick a7b370ce26e528796f61f0ebb64dd9e76507e92b.

**Comment by yuyongyang800** [comment]
Done! This cherry-pick commit is pushed to the master branch.

**Comment by merlintang** [comment]
thanks!!!

**Pull request merged**